core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -250,13 +250,13 @@ private[deploy] class Master(
       if (state == RecoveryState.STANDBY) {
         workerRef.send(MasterInStandby)
       } else if (idToWorker.contains(id)) {
-        workerRef.send(RegisterWorkerFailed("Duplicate worker ID"))
+        workerRef.send(RegisteredWorker(self, masterWebUiUrl, masterAddress, true))
```

Comment:
Do we have to send back the signal to the worker? Can we just log warnning here and reply nothing?

Reply:
Yes, we have to. Each time worker is trying to connect to masters, it sets `registered=false` and  only set to `true` when it receives master's response(`RegisteredWorker`).